src/Users/Controllers/UserController.php

```diff
@@ -157,8 +157,16 @@ public function save(?int $userId = null)
         // Check for an avatar to upload
         if ($file = $this->request->getFile('avatar')) {
             if ($file->isValid()) {
+              
+                $avatar_dir = ROOTPATH . 'public/uploads/avatars';
                 $filename = $user->id . '_avatar.' . $file->getExtension();
-                if ($file->move(ROOTPATH . 'public/uploads/avatars', $filename, true)) {
+                
+                // Create if uploads/avatar directories not exist 
+                if (!is_dir($avatar_dir)){
+                    mkdir($avatar_dir, 0777, true);
```

Comment:
IIRC, many shared hosts have issues with folders being 777 permissions. I think 0755 should be fine, though. 

```suggestion
                    mkdir($avatar_dir, 0755, true);
```